--- a/src/unstoppable/UnstoppableVault.sol
+++ b/src/unstoppable/UnstoppableVault.sol
@@ -82,7 +82,7 @@
         if (amount == 0) revert InvalidAmount(0); // fail early
         if (address(asset) != _token) revert UnsupportedCurrency(); // enforce ERC3156 requirement
         uint256 balanceBefore = totalAssets();
-        if (convertToShares(totalSupply) != balanceBefore) revert InvalidBalance(); // enforce ERC4626 requirement
+        if (convertToAssets(totalSupply) != balanceBefore) revert InvalidBalance(); // enforce ERC4626 requirement
 
         // transfer tokens out + execute callback on receiver
         ERC20(_token).safeTransfer(address(receiver), amount);
@@ -131,4 +131,4 @@
         if (flag) _pause();
         else _unpause();
     }
-}
+}